src/mlpack/methods/ann/augmented/tree_memory_impl.hpp

```diff
@@ -12,18 +12,22 @@
 #ifndef MLPACK_METHODS_AUGMENTED_TREE_MEMORY_IMPL_HPP
 #define MLPACK_METHODS_AUGMENTED_TREE_MEMORY_IMPL_HPP
 
+#include <mlpack/methods/ann/layer/layer.hpp>
```

Comment:
We should include the header file in ``tree_memory.hpp`` so that ``LayerTypes`` is known. Also, currently the Constructor definition doesn't match with the implementation; looks like there are two options using ``FFN`` or ``LayerTypes``, if you go for ``FFN`` you don't have to use the visitor pattern, but on the other side you are restricted to a forward model, so passing a recurrent model isn't something we could do. think. I think either FFN or LayerTypes is fine here, let me know what you think.